DiagnosticsExtension/Models/ConnectionStringValidator/BlobStorageValidator.cs

```diff
@@ -73,46 +71,39 @@ public async Task<TestConnectionData> TestConnectionStringViaAppSetting(string a
-                    if (string.IsNullOrEmpty(serviceUriString))
-                    {
-                        serviceUriString = Environment.GetEnvironmentVariable(appSettingName + "__serviceUri");
-                    }
-                    if (!string.IsNullOrEmpty(serviceUriString))
+                    serviceUriString = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.ServiceUri);
+                }
+                if (!string.IsNullOrEmpty(serviceUriString))
+                {
+                    appSettingClientIdValue = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.ClientId);
+                    appSettingClientCredValue = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.Credential);
+                    Uri serviceUri = new Uri(serviceUriString);
+                    // Creating client using User assigned managed identity
+                    if (!string.IsNullOrEmpty(appSettingClientIdValue))
                     {
-                        appSettingClientIdValue = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.ClientId);
-                        appSettingClientCredValue = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.Credential);
-                        Uri serviceUri = new Uri(serviceUriString);
-                        // Creating client using User assigned managed identity
-                        if (!string.IsNullOrEmpty(appSettingClientIdValue))
+                        if (appSettingClientCredValue != ConnectionStringResponseUtility.ValidCredentialValue)
                         {
-                            if (appSettingClientCredValue != ConnectionStringResponseUtility.ValidCredentialValue)
-                            {
-                                throw new ManagedIdentityException(ConnectionStringResponseUtility.ManagedIdentityCredentialMissing);
-                            }
-                            else
-                            {
-                                identityType = ConnectionStringValidationResult.ManagedIdentityType.User;
-                                client = new BlobServiceClient(serviceUri, new Azure.Identity.ManagedIdentityCredential(appSettingClientIdValue));
-                            }
+                            throw new ManagedIdentityException(ConnectionStringResponseUtility.ManagedIdentityCredentialMissing);
                         }
-                        // Creating client using System assigned managed identity
                         else
                         {
-                            identityType = ConnectionStringValidationResult.ManagedIdentityType.System;
-                            client = new BlobServiceClient(serviceUri, new Azure.Identity.ManagedIdentityCredential());
+                            response.IdentityType = ConnectionStringResponseUtility.User;
```

Comment:
can you include this in the TestConnectionData that is returned by this method instead of having a class member?  Response is not really a class member and is being used to communicate return a return value from one function to another - this is a bad coding pattern and something I raised in the last two rounds of PR feedback but don't see addressed.

Reply:
fixed